spec/controllers/admin/application_controller_spec.rb

```diff
@@ -0,0 +1,43 @@
+require "rails_helper"
+
+module Admin
+  class OrdersController < Admin::ApplicationController
+    def after_resource_destroyed_path(_requested_resource)
+      { action: :index, controller: :customers }
+    end
+
+    def after_resource_created_path(requested_resource)
+      [namespace, requested_resource.customer]
+    end
+
+    def after_resource_updated_path(requested_resource)
+      [namespace, requested_resource.customer]
+    end
+  end
+end
```

Comment:
Creating modules like this in spec files can create issues, as the modules stay around afterwords, leaving the state modified. For example, a spec that runs after this one would get this implementation of the hooks.

A better way to implement this is to use RSpec's "anonymous controller" feature, which allows you to create a disposable child class of the controller you are testing: https://relishapp.com/rspec/rspec-rails/docs/controller-specs/anonymous-controller

Reply:
@pablobm , thanks a lot for the explanation, really nice!